third_party/closure-compiler/runner/src/AmpCommandLineRunner.java

```diff
@@ -0,0 +1,34 @@
+import java.util.List;
+
+import com.google.javascript.jscomp.CommandLineRunner;
+import com.google.javascript.jscomp.CompilerOptions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+public class AmpCommandLineRunner extends CommandLineRunner {
+
+  private static final List<String> LOG_STRIP_SUFFIX = ImmutableList.of(
+      "dev.fine",
+      "dev.assert");
+
+  protected AmpCommandLineRunner(String[] args) {
+    super(args);
+  }
+
+  @Override
+  protected CompilerOptions createOptions() {
+    CompilerOptions options = super.createOptions();
+    options.setStripTypes(Sets.newHashSet(LOG_STRIP_SUFFIX));
```

Comment:
We have 2 mechanisms to do DCE, `setStripTypes/setStripTypePrefixes` (option A) and `setStripNameSuffixes/setStripNamePrefixes` (option B). option A is more safer as we can declare an expression like `dev.fine` while option b seems to only allow `fine` or `assert` (no access operation, just method name).

the issue is that the strip operation seems to occur **after** the `SIMPLE_PLUS_OPTIMIZATION` has ran, meaning if you did `setStripTypes('dev.fine')` (option A), it wouldn't work, as the actual js code it operates on would have be `a.fine` (after var renaming). Now we could use option B, but option B is extremely unsafe since `fine` or `assert` are very generic and any other module/class could declare one, and to get around this we would need the `log` module to declare method names like `devfine` and `devassert` (even then it feels extremely unsafe).

One other option we can do is assign the log instances to the global `AMP` object since that is never renamed and an expression like `AMP.dev.fine('hello world')` would be stripped out if we do `setStripTypes('AMP.dev.fine'). But this would also mean `log` has to be the very first module loaded since it would be needed by the other modules that uses logging.

/cc @cramforce @dvoytenko